lib/v2/elsevier/volume.js

```diff
@@ -0,0 +1,83 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const path = require('path');
+const { art } = require('@/utils/render');
+
+module.exports = async (ctx) => {
+    const resp = await got('https://www.sciencedirect.com');
+    const cookie = resp.headers['set-cookie'].join(' ');
+    const journal = ctx.params.journal;
+    const volume = 'Volume ' + ctx.params.id.replace('-', ' Issue ');
+    const volUrl = 'https://www.sciencedirect.com/journal/' + journal + '/vol/' + ctx.params.id.replace('-', '/issue/');
+
+    const volResponse = await got({
+        method: 'get',
+        url: volUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
+            Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+            Referer: 'https://id.elsevier.com/',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept-Language': 'zh-CN,zh;q=0.9',
+            cookie,
+        },
+    });
+
+    const $ = cheerio.load(volResponse.data);
+    const jrnlName = $('.anchor.js-title-link').text();
+
+    const list = $('.js-article')
+        .map((_, item) => {
+            const title = $(item).find('.js-article-title').text();
+            const authors = $(item).find('.js-article__item__authors').text();
+            const link = $(item).find('.article-content-title').attr('href');
+            const id = $(item).find('.article-content-title').attr('id');
+            return {
+                title,
+                link,
+                id,
+                authors,
+                volume,
+            };
+        })
+        .get();
+
+    const renderDesc = (item) =>
+        art(path.join(__dirname, 'templates/description.art'), {
+            item,
+        });
+
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const itemResponse = await got({
+                    method: 'get',
+                    url: 'https://www.sciencedirect.com/science/article/pii/' + item.id,
+                    headers: {
+                        Host: 'www.sciencedirect.com',
```

Comment:
Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `url`?

Reply:
Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```